ClickHouse.Client.Tests/Misc/UriBuilderTests.cs

```diff
@@ -29,8 +33,129 @@ public void ShouldSetUriParametersCorrectly()
         Assert.AreEqual("SELECT 1", @params.Get("query"));
         Assert.AreEqual("1", @params.Get("a"));
         Assert.AreEqual("c", @params.Get("b"));
+        Assert.AreEqual("1", @params.Get("c"));
+        Assert.AreEqual("c", @params.Get("d"));
         Assert.AreEqual("SESSION", @params.Get("session_id"));
         Assert.AreEqual("false", @params.Get("enable_http_compression"));
+        Assert.AreEqual("QUERY", @params.Get("query_id"));
+        Assert.AreEqual("sqlParameterValue", @params.Get("param_sqlParameterName"));
+    }
+
+    [Test]
+    public void CommandQueryStringParametersShouldOverrideConnectionParameters()
+    {
+        var builder = new ClickHouseUriBuilder(new Uri("http://some.server:123"))
+        {
+            ConnectionQueryStringParameters = new Dictionary<string, object> { { "a", 1 } },
+            CommandQueryStringParameters = new Dictionary<string, object> { { "a", 2 } },
+        };
+
+        var result = new Uri(builder.ToString());
+        var @params = HttpUtility.ParseQueryString(result.Query);
+
+        Assert.AreEqual("2", @params.Get("a"));
+    }
+
+    [Test]
+    public void ConnectionQueryStringParametersShouldOverrideCommonParameters()
+    {
+        var builder = new ClickHouseUriBuilder(new Uri("http://some.server:123"))
+        {
+            Database = "DATABASE",
+            UseCompression = false,
+            Sql = "SELECT 1",
+            SessionId = "SESSION",
+            QueryId = "QUERY",
+            ConnectionQueryStringParameters = new Dictionary<string, object>
+            {
+                { "database", "overrided" },
+                { "enable_http_compression", "overrided" },
+                { "query", "overrided" },
+                { "session_id", "overrided" },
+                { "query_id", "overrided" },
+            },
+        };
+
+        builder.AddSqlQueryParameter("sqlParameterName", "sqlParameterValue");
+
+        var result = new Uri(builder.ToString());
+        var @params = HttpUtility.ParseQueryString(result.Query);
+
+        Assert.AreEqual("overrided", @params.Get("database"));
+        Assert.AreEqual("overrided", @params.Get("enable_http_compression"));
+        Assert.AreEqual("overrided", @params.Get("query"));
+        Assert.AreEqual("overrided", @params.Get("session_id"));
+        Assert.AreEqual("overrided", @params.Get("query_id"));
```

Comment:
I don't think it's a correct behavior design-wise to be able to override 'system' fields like these. There is other behavior tied into these flags, so overriding them will lead to unexpected behavior. What is your idea behind allowing to override these?

Reply:
That's fair, I'll need to change that separately